### ISSUE TITLE
Dataproject – Feedback from Valros

### ISSUE THREAD

**Issue by xyzxyz-gvz** [NumEconCopenhagen/projects-2023-git#3]
Overall: 
Very cool project:) I found the use of random forest to be quite nice and interesting that you managed to get a prediction. The only thing that I would recommend would be to make the code a bit more readable – as some of the long code modules could have been made into a function and imported from a py-file. All in all very nice, now if you will excuse me, I have some betting to do. 

**The most elegant solution in the project was:** The use of the Poisson distribution to calculate the probabilities of home and away teams scoring different numbers of goals in a match. This approach provides an intuitive and statistically sound way of representing match outcomes.
 
**The hardest section of the code in the project to understand was:** The new model using logistic regression to predict the probability of the home team winning for each match, and then using the Poisson distribution to predict the number of goals. The random nature of the Poisson distribution and the absence of a fixed seed made the results inconsistent – I see you have commented on that as well. 
 
**This part of the project could be better documented:** The creation of the 'upcoming_matches' data frame and how it's used for making predictions. More explanation of the columns and the data could make it easier for readers to follow the code and understand the purpose of this data frame. Also since you choose to use lambda functions. 
 
**An idea for an improvement/clarification could be:** An additional improvement and clarification, would be to make to code more readable. It’s very hard on the eyes and you really have to concentrate. I believe this would make your code much easier to understand and read as well. 

An idea for an extension could be: Incorporating additional features into the models, such as team form, injuries, and other situational factors. This would allow the models to provide more accurate predictions by considering a wider range of factors affecting match outcomes. Additionally, you could experiment with different machine learning models, such as XGBoost or neural networks, to find the most suitable model for this classification prediction task.
